app/models/miq_alert_status_action.rb

```diff
@@ -0,0 +1,30 @@
+class MiqAlertStatusAction < ApplicationRecord
```

Comment:
[It's easy to mark a relation as read-only](http://api.rubyonrails.org/v5.0.0.1/classes/ActiveRecord/QueryMethods.html#method-i-readonly), but read only implies you can't update *or delete* records. I believe you'll have to utilize callbacks as @kbrock mentions if you want to allow deletes but not updates.